File: compiler.js
const fs = require('fs');
const path = require('path');

const inputFile = path.join(__dirname, 'linguistic_analysis.json');
const outputJsFile = path.join(__dirname, 'src', 'lessonData.js');

if (!fs.existsSync(inputFile)) {
    console.error("linguistic_analysis.json not found!");
    process.exit(1);
}

const data = JSON.parse(fs.readFileSync(inputFile, 'utf-8'));

// If a src directory exists (e.g. Vite/React template), inject data as a JS module
if (fs.existsSync(path.join(__dirname, 'src'))) {
    const jsContent = `export const LESSON_DATA = ${JSON.stringify(data, null, 2)};`;
    fs.writeFileSync(outputJsFile, jsContent, 'utf-8');
    console.log("Injected data into src/lessonData.js");
} else {
    // If not, just create a standalone JS file that defines window.LESSON_DATA
    const jsContent = `window.LESSON_DATA = ${JSON.stringify(data, null, 2)};`;
    fs.writeFileSync(path.join(__dirname, 'lessonData.js'), jsContent, 'utf-8');
    console.log("Created lessonData.js (window.LESSON_DATA)");
}

File: compiler.py
import os
import json

def compile_html():
    json_path = "linguistic_analysis.json"
    template_path = "template.html"
    output_path = "shadowing_course.html"

    if not os.path.exists(json_path):
        print(f"{json_path} not found.")
        return

    if not os.path.exists(template_path):
        print(f"{template_path} not found. Please provide the template.html file.")
        return

    with open(json_path, "r", encoding="utf-8") as f:
        data = json.load(f)

    with open(template_path, "r", encoding="utf-8") as f:
        html_content = f.read()

    # Inject data into <head>
    data_script = f"<script>window.LESSON_DATA = {json.dumps(data, ensure_ascii=False)};</script>\n"
    
    # We find the </head> tag and insert right before it
    if "</head>" in html_content:
        html_content = html_content.replace("</head>", data_script + "</head>")
    else:
        # Fallback if no head tag is found
        html_content = data_script + html_content

    with open(output_path, "w", encoding="utf-8") as f:
        f.write(html_content)

    print(f"Successfully compiled to {output_path}")

if __name__ == "__main__":
    compile_html()
